autoload/neomake.vim

```diff
@@ -368,9 +371,9 @@ function! s:AddExprCallback(maker) abort
 
     if list_modified
         if file_mode
-            call setloclist(a:maker.winnr, list, 'r')
+            call setloclist(a:maker.winnr, list, 'r', name)
```

Comment:
Testing how this is exercised in neomake. There are two places where I changed the code. I'm not sure when `list_modified` is true in this case. The first change gets executed in my `Ag` command:

```
function! s:Ag(file_mode, args)
  let cmd = "ag --vimgrep --smart-case ".substitute(a:args, '\\', '\\\\', 'g')
  let custom_maker = neomake#utils#MakerFromCommand('bash', cmd)
  let custom_maker.name = cmd
  let custom_maker.remove_invalid_entries = 0
  let custom_maker.place_signs = 0
  let custom_maker.errorformat = "%f:%l:%c:%m"
  let enabled_makers =  [custom_maker]
  call neomake#Make(a:file_mode, enabled_makers) | echo "running: " . cmd
endfunction
command! -bang -nargs=* -complete=file Ag call s:Ag(<bang>0, <q-args>)
```